src/scoringreport.rs

```diff
@@ -0,0 +1,16 @@
+use std::fs;
+use crate::{config::Check};
+
+/// Writes to Scoring Report
+pub fn write_to_scoring_report (file: &String, scored_vector: impl Iterator<Item = Check>) {
+  let mut text = String::from("<ul>\n");
+  for scored in scored_vector {
```

Comment:
instead of using a for loop you could use [map](https://doc.rust-lang.org/std/iter/trait.Iterator.html#method.map) to transform it accordingly